src/frontend/src/components/deviceListItem.ts

```diff
@@ -1,17 +1,32 @@
 import { TemplateResult, html } from "lit-html";
 import { Setting, settingName } from "../flows/manage/deviceSettings";
-import { warningIcon, dropdownIcon, lockIcon } from "./icons";
+import {
+  warningIcon,
+  errorIcon,
+  successIcon,
+  dropdownIcon,
+  lockIcon,
+} from "./icons";
 
 // A simple representation of "device"s used on the manage page.
 export type Device = {
   // All the settings allowed for a particular device
-  settings: Setting[];
+  settings?: Setting[];
```

Comment:
is this needed? looks like empty `[]` is used in the code when there is no settings

Reply:
yes. when we want you to add a new device, there are no settings passed. Do you think its nicer to alway pass empty settings everywhere?